File: NTH_Restaurant_API/src/main/java/com/example/NTH_Restaurant_API/entity/MonAnEntity.java
package com.example.NTH_Restaurant_API.entity;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import javax.validation.constraints.Size;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "MONAN")
public class MonAnEntity {
    @Id
    @Column(name = "MAMA")
    @Size(max = 10)
    private String maMA;

    @Column(name = "TENMA")
    @Size(max = 50)
    private String tenMA;

    @Column(name = "CHUTHICH")
    @Size(max = 255)
    private String chuThich;

    @Column(name = "TRANGTHAI")
    @Size(max = 50)
    private String trangThai;

    @Column(name = "HINHANH")
    @Size(max = 255)
    private String hinhAnh;

    @OneToMany(mappedBy = "mama")
    private List<CT_ThucDonEntity> ctThucdonList;

    @OneToMany(mappedBy = "mama")
    private List<CT_DatMonEntity> ctDatmonList;

    @OneToMany(mappedBy = "mama")
    private List<CT_MonAnEntity> ctMonanList;

    @OneToMany(mappedBy = "mama")
    private List<ThayDoiGiaMonEntity> thaydoigiamonList;

    @OneToMany(cascade = CascadeType.REFRESH, mappedBy = "mama")
    private List<CT_DatMonTruocEntity> ctDatmontruocList;
    @ManyToOne
    @JoinColumn(name = "MALMA")
    private LoaiMonAnEntity malma;
}

File: NTHRestaurant/app/src/main/java/com/example/nthrestaurant/network/model/MonAnEntity.kt
package com.example.nthrestaurant.network.model

import com.squareup.moshi.Json

data class MonAnEntity(
    @Json(name = "chuThich") val chuThich: String?,
    @Json(name = "gia") val gia: Int,
    @Json(name = "hinhAnh") val hinhAnh: String,
    @Json(name = "maMA") val maMA: String,
    @Json(name = "malma") val maLMA: String,
    @Json(name = "tenMA") val tenMA: String,
    @Json(name = "tenlma") val tenLMA: String,
    @Json(name = "trangThai") val trangThai: String
)